### PR TITLE
FEATURE: add the 202401 review blog and fix the template of blog

### PR DESCRIPTION
**Pull request by kuomc**
In this PR, I add the content of 202401 review blog. I also corrected the layout of the [blog page](https://sciwork.dev/blog). But I am not sure if this is a good way.

### CODE REVIEW

**Comment by kuomc** [comment]
Hi @yungyuc , can you help me check if the content is appropriate?

**Comment by kuomc** [comment]
image
This is the original blog layout. The height of two blocks in a row are different. I add the flex and flex-col class in each blog block. Then, I add a div with flex-1 class grow and fill the available space to make sure the size of each blog block are the same. But I am not sure if this is a good method. I thought I could use the appropriate class to achieve the goal, but I can't find it.

File: theme/templates/blog.html [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <div class="flex flex-row flex-wrap -mx-2">
       {% for article in articles_page.object_list %}
         <div class="w-full sm:w-full md:w-1/2 mb-4 px-2">
-          <div class="max-w-4xl px-10 my-4 py-6 bg-white rounded-lg shadow-md">
+          <div class="max-w-4xl flex flex-col h-full px-10 my-4 py-6 bg-white rounded-lg shadow-md">
```
Comment (rockleona) [inline theme/templates/blog.html:10]: Just being curious about the layout, I've tried comparing the change before (like the code below) and after, 
but I found previous version it works fine on my screen. 
```suggestion
          <div class="max-w-4xl h-full px-10 my-4 py-6 bg-white rounded-lg shadow-md">
```

<details>
<summary>Screenshots</summary>

image

image

</details>

Comment (rockleona) [reply theme/templates/blog.html:10]: Oh, I've tried to shorten the blog article, which can see the difference now.

<details>
<summary>Screenshots</summary>

image

</details>

File: content/blog/2024/202401-review.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+================================================================
+
+:date: 2024-03-04
+:author: Ming-Jie Kuo
+:category: blog
+
+Weekly Meetup
+-------------
+In the January meetup, Yung-Yu introduced the CESE method, which is utilized in
+the modmesh project to solve the fluid dynamics problem. This provided modmesh
```
Comment (yungyuc) [inline content/blog/2024/202401-review.rst:12]: You may add the link to the modmesh repository as a passive way to promote, if you want.

File: content/blog/2024/202401-review.rst [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+members with a deeper understanding of using numerical methods to solve the partial
```
Comment (yungyuc) [inline content/blog/2024/202401-review.rst:13]: I recommend to say `s/members/contributors/`

File: content/blog/2024/202401-review.rst [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+differential equations. Meanwhile, Ming-Jie delved into researching the CESE method.
+In addition, Kuan-Wei Chiu and Chun-Shih, former students of the nsd course
```
Comment (yungyuc) [inline content/blog/2024/202401-review.rst:15]: Please do not make contributor background explicit, unless you are asked to do so, or it is necessary.  It's to respect other's privacy.

**Review by yungyuc — CHANGES_REQUESTED** [review]
- [ ] Be conservative in writing when it may have anything to do with others' privacy

File: content/blog/2024/202401-review.rst (hunk 2 above)
Comment (kuomc) [reply content/blog/2024/202401-review.rst]: I added a link to the modmesh repository.

File: content/blog/2024/202401-review.rst (hunk 4 above)
Comment (kuomc) [reply content/blog/2024/202401-review.rst]: I removed the contributors' background.

File: content/blog/2024/202401-review.rst (hunk 3 above)
Comment (kuomc) [reply content/blog/2024/202401-review.rst:13]: Do you mean "This provided modmesh contributors with …"?

Comment (yungyuc) [reply content/blog/2024/202401-review.rst:13]: Yes

**Review by yungyuc — APPROVED** [review]

**Pull request merged**